### ISSUE TITLE
[Question] Does UCR support touchpad swipes?

### ISSUE THREAD

**Issue by gaveitatry2** [Snoothy/UCR#213]
ViGEmBus 1.17 supports touchpad swipes, so I am trying to find some programs that have that implemented.  Have you implemented touchpad swipes yet?

Also, I have another question.  You have a long page of information about what UCR does, but it's really confusing to me.  It's like trying to read something in another language that I'm not familiar with.  Can you dumb it down for me?  If I launch UCR, will PlayStation Now automatically detect UCR as a PS4 Controller?  And will UCR let me use my keyboard and mouse instead of a PS4 Controller and let me assign keyboard keys and mouse buttons to the various PS4 Controller buttons like triangle, square, cross, and circle?

**Comment by DanTDBV** [comment]
I wrote this tutorial that shows some of UCR the basics. I have not gone into the deeper levels, but it is a starting point.

https://www.steelbeasts.com/topic/13639-how-to-using-multiple-controllers-with-steel-beasts-pro-pe/

I hope it can get the ball rolling for you.

Good luck.

**Comment by gaveitatry2** [comment]
I don't know why you think that tutorial will help.  I already stated that I don't have a controller.  Your tutorial is on how to make the most out of controllers you own.  I don't see the connection.

**Comment by gaveitatry2** [comment]
I'm just going to assume UCR isn't the program I need.  On Feb 6th, I found the GitHub page for DS4Emulator and that's what I've been using ever since.  It's an excellent program.  Easy to understand instructions and easy to use program and does everything excellently.  But it hasn't been updated to do touchpad swipes yet.  And, it could be even better if it had it's own change controls window instead of having to use a text config file to change the controls.

**Comment by WelpThatWorked** [comment]
UCR doesn't have touchpad support currently.

If you use ViGEm, the virtual devices are created when you activate the profile, so once you do, Playstation Now will detect them as if they were a real controller.
